### PR TITLE
Scrollable Promo Data

### LINKED ISSUE #9697: Editorial onward journey component: horizontally scrollable promos
**Is your feature request related to a problem? Please describe.**
As part of our objective to scale Optimo publishing to editorial users, we want to support the editorial teams in their ability to link to other stories, videos or content that are relevant to the narrative or provide deeper context, understanding and analysis around a story.

We would like to build an editorial onward journey component for a set of links for Optimo articles.

**Describe the solution you'd like**
A new component that consumes Optimo data and renders a maximum of 3 link promos with event tracking.

<details>
<summary>Background</summary>

We want to enable editorial users to share links within the body of an article without completely disrupting the user flow.

There will be an editorially curated “Links” block within Optimo that can be blended with other onward journeys in the future. 

The "Links" block only contains a set of 'Link' types containing text and images only

This block will serve a similar purpose to both the see-alsos and the bulleted hyperlinks in stories.

Editorial can place the block or blocks anywhere within the article. There are no limits on now many can be used.

The blocks will be considered as part of the article and the  components are rendered as part of the article

The links block can have titles set in Optimo

There will be 2 components powered by this block on the page : - 'more on this story' and 'Related content'.

Test Optimo (Optimo work in progress. This is shown as example)

Link to article with `links` blocks: https://www.test.bbc.com/pidgin/articles/crrrkxz2k0ko.json

</details>

<details>
<summary>Design</summary>

Screenshot 2022-01-12 at 11 57 42

Refer to NEWSWORLDSERVICE-1223 for more info on designs

</details>

**Developer notes**
- Develop with RTL/LTR languages in mind and ensure horizontal scroll scrolls in correct direction for languages
- Render null if no promos are contained in the block

**Questions**
- Should we split event tracking into separate issue? (DR: Yes)
- Do we use CSS line-clamp for truncating to 3 lines? This isn't supported in older browsers but this may be ok to let these lines flow to +3 lines in older browsers (DR: It is ok for older browser to flow 3+ lines)
- Ask UX about the width of promo when links are horizontally scrollable e.g. set amount of pixels or percentage of viewport? (DR: set amount of pixels at different viewports)
- Do we have a minimum promo box size for very small devices? (DR: No for 2 reasons: Might be an edge case we don't care about and because this is mitigated by the scrollable nature of the component)

**a11y notes**
- Please refer to NEWSWORLDSERVICE-1223 for up to date a11y acceptance criteria

**Testing notes**
Test page available 
https://www.test.bbc.com/pidgin/articles/crrrkxz2k0ko
https://www.test.bbc.com/pidgin/articles/crrrkxz2k0ko.json

RtoL
https://www.test.bbc.com/arabic/articles/cjn2e0jek55o
https://www.test.bbc.com/arabic/articles/cjn2e0jek55o.json

- Test mobile and desktop
- Test RTL/LTR
- Test truncation
- Test number of promos to test the 3 maximum e.g. test with 2 promos, test with +3 promos
- Test tabbing order
- Test viewport size and promo comes into view when tabbed to

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [x] This feature is expected to need manual testing.

**Checklist**
- [x] (BBC contributors only) This issue follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Additional context**

### PR DESCRIPTION
**Pull request by rebeccamcginn**
Resolves #9697

**Overall change:**
Checks if last block in article content is of type 'links'. If so, changes this type to 'relatedContent'. Links blocks that are destined for the scrollable promo remain of type 'links'.

**Code changes:**

- _A bullet point list of key code changes that have been made._
- _When describing code changes, try to communicate **how** and **why** you implemented something a specific way, not just **what** has changed._

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh project
- [ ] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/app/routes/article/handlePromoData/index.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+const handlePromoData = pageData => {
+  const dataBlocks = getBlocks(pageData);
+  const finalBlock = dataBlocks.length - 1;
+
+  if (dataBlocks[finalBlock].type === 'links') {
+    dataBlocks[finalBlock] = assocPath(
+      ['type'],
+      'relatedContent',
+      dataBlocks[finalBlock],
+    );
```
Comment (ryanmccombe) [inline src/app/routes/article/handlePromoData/index.js]:
```suggestion
    dataBlocks[finalBlock].type = 'relatedContent';
```

**Review by HarveyPeachey — APPROVED** [review]
Nice and Clean. So is this `relatedContent` type meant to be a replacement of the `relatedContent` found at the [end](https://github.com/bbc/simorgh/blob/268835666734f523530c0df27059217025d4d8e2/data/pidgin/articles/crrrkxz2k0ko.json#L1139) of the asset?

**Comment by rebeccamcginn** [comment]
> Nice and Clean. So is this `relatedContent` type meant to be a replacement of the `relatedContent` found at the [end](https://github.com/bbc/simorgh/blob/268835666734f523530c0df27059217025d4d8e2/data/pidgin/articles/crrrkxz2k0ko.json#L1139) of the asset?

Negative. The blocks for the scrollable promo and for related content are both of type 'links'. We need to differentiate between them so that the links block intended for the related content component is not rendered by the scrollable promo component. 

We are using the links type here for the scrollable promo: https://github.com/bbc/simorgh/blob/268835666734f523530c0df27059217025d4d8e2/src/app/pages/ArticlePage/ArticlePage.jsx#L79

We are changing the type of the last links block to relatedContent - if it is the last block in content.model.blocks - so that it is not rendered as a scrollable promo: https://github.com/bbc/simorgh/blob/268835666734f523530c0df27059217025d4d8e2/data/pidgin/articles/crrrkxz2k0ko.json#L807

Then, back in ArticlePage, we will use the new type 'relatedContent' for rendering the relatedContent promo

**Comment by rebeccamcginn** [comment]
I can see having two `relatedContents` might get confusing so we could change the name of our new one. `relatedContentPromo` perhaps.

**Comment by HarveyPeachey** [comment]
> I can see having two `relatedContents` might get confusing so we could change the name of our new one. `relatedContentPromo` perhaps.

Yeah that makes sense. I guess it confused me a little because I'm not sure what the difference is between them currently

**Pull request merged**